api/openapi/model-registry.yaml

```diff
@@ -970,36 +1093,39 @@ components:
       type: string
     ModelArtifact:
       description: An ML model artifact.
-      type: object
-      required:
-        - artifactType
-      properties:
-        artifactType:
-          type: string
-          default: "model-artifact"
       allOf:
         - $ref: "#/components/schemas/BaseArtifact"
         - $ref: "#/components/schemas/ModelArtifactCreate"
     DocArtifact:
       description: A document.
       type: object
+      allOf:
+        - $ref: "#/components/schemas/BaseArtifact"
+        - $ref: "#/components/schemas/DocArtifactCreate"
+    DocArtifactCreate:
+      description: A document artifact to be created.
+      type: object
+      allOf:
+        - $ref: "#/components/schemas/BaseArtifactCreate"
+        - $ref: "#/components/schemas/DocArtifactUpdate"
+    DocArtifactUpdate:
+      description: A document artifact to be updated.
       required:
         - artifactType
-      properties:
-        artifactType:
-          type: string
-          default: "doc-artifact"
       allOf:
-        - $ref: "#/components/schemas/BaseArtifact"
+        - $ref: "#/components/schemas/BaseArtifactUpdate"
+        - type: object
+          properties:
+            artifactType:
+              type: string
+              default: "doc-artifact"
```

Comment:
Have you tried what happens if you don't make this changes, how does it impact the .patch below?

saying because iirc with @lampajr we tried several "places" where to put the discriminator, and that indeed impacted on the way the codegen behaved for this attribute

Reply:
the request fails, the current properties setup outside the AllOf block is invalid iiuc

Reply:
>the current properties setup outside the AllOf block is invalid iiuc

weird because that's exactly what was making the auto-generation working - but I don't know what was the culprit.. either the generator that is broken (not surprising) or the openapi spec that was wrong

Reply:
> weird because that's exactly what was making the auto-generation working

I recall we had to move these discriminator around in a few places to indeed get the openapi-codegen working, and ultimately we settled for a "convention" on how to make the discriminator being generated as expected.

That's also to be added that new stuff is being added in this PR, so I would also not exclude that by different set of entities/endpoint, that "convention" no longer holds? 🤷 

I'm happy to hear however @lampajr we did recall the same thing!

Reply:
I'll review that because I remember just later yesterday I found an issue with the docker builder. So I might have tested stale versions.
TBH I don't remember the generation being different in any case. However this usage of `allOf` is what's listed on they swagger docs.

Reply:
Looks like this actually works, I'll update it soon.